code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -24,7 +24,11 @@
 
 	if(!S) return
 
-	chambered.BB = new /obj/item/projectile/bullet/dart/syringe(src)
+	if(istype(S, /obj/item/weapon/reagent_containers/syringe/piercing))
```

Comment:
give ```S``` (the type of syringe)``` a variable called ```projectile_type```, so it's like:
```
/obj/item/weapon/reagent_containers/syringe
   ...
   var/projectile_type = /obj/item/projectile/bullet/dart/syringe

/obj/item/weapon/reagent_containers/syringe/piercing
   ...
   projectile_type = /obj/item/projectile/bullet/dart/syringe/piercing
```

then here you can do
```chambered.BB = new S.projectile_type```